### DISCUSSION TITLE
How does one make a vscode extension

### DISCUSSION THREAD

**Discussion by wunadacatt** [microsoft/vscode-discussions#630] [Extension Development QnA] [answered]
Im trying to make a custom vscode extension for a language that was custom made. I would like it to include style and code completion. How would one manage that

**Comment by apupier** [comment, marked as answer]
You can look to the first extension doc page https://code.visualstudio.com/api/get-started/your-first-extension for the generic needs.

and then there is a specific documentation for Language extension https://code.visualstudio.com/api/language-extensions/overview
